spinalcordtoolbox/process_seg.py

```diff
@@ -78,30 +78,30 @@ def compute_shape(segmentation, angle_correction=True, centerline_path=None, par
         else:
             im_centerline_r = im_segr
         # compute the spinal cord centerline based on the spinal cord segmentation
-        # here, param_centerline.minmax needs to be False because we need to retrieve the total number of input slices
         _, arr_ctl, arr_ctl_der, fit_results = get_centerline(im_centerline_r, param=param_centerline, verbose=verbose,
                                                               remove_temp_files=remove_temp_files)
+        # the third column of `arr_ctl` contains the integer slice numbers, and the first two
+        # columns of `arr_ctl_der` contain the x and y components of the centerline derivative
+        deriv = {int(z_ref): arr_ctl_der[:2, index] for index, z_ref in enumerate(arr_ctl[2])}
 
     # Loop across z and compute shape analysis
     for iz in sct_progress_bar(range(min_z_index, max_z_index + 1), unit='iter', unit_scale=False, desc="Compute shape analysis",
                                ncols=80):
         # Extract 2D patch
         current_patch = im_segr.data[:, :, iz]
         if angle_correction:
+            try:
+                dx, dy = deriv[iz]
+            except KeyError:
+                raise ValueError(f"The provided angle correction centerline does not cover slice {iz} of "
+                                 "the input mask. Please supply a more extensive -angle-corr-centerline, "
+                                 "or set -angle-corr to 0.") from None
```

Comment:
Very minor nit: 

```suggestion
                raise ValueError(f"The provided angle correction centerline does not cover slice {iz} of "
                                 "the input mask. Please supply a more extensive '-angle-corr-centerline', "
                                 "or disable angle correction ('-angle-corr 0').") from None
```

Just a tad clearer on the face of it IMO, though I'm not married to this change if you disagree.

(Also, TIL you can suppress the stack with `from None`; good to know, will be using that from now on)